lib/server.js

```diff
@@ -275,6 +275,17 @@ function create(log, error, config, routes, db, translator) {
 
       defineLazyGetter(request.app, 'ua', () => userAgent(request.headers['user-agent']))
       defineLazyGetter(request.app, 'geo', () => getGeoData(request.app.clientAddress))
+      defineLazyGetter(request.app, 'devices', () => {
+        let uid
+
+        if (request.auth && request.auth.credentials) {
+          uid = request.auth.credentials.uid
+        } else if (request.payload && request.payload.uid) {
+          uid = request.payload.uid
+        }
+
+        return db.devices(uid)
```

Comment:
Initially I had this resolving to an empty array if we don't have a `uid`. But I realised that's wrong, this property should behave identically to a call to `db.devices`, including rejecting if the `uid` is bad.

Not that I expect it to get invoked without a `uid`, but if it does it should definitely fail.